.husky/pre-commit

```diff
@@ -0,0 +1,2 @@
+npm run test:generation
```

Comment:
What does this do?

Reply:
Some contracts are automatically generated with the `npm run generate` command, we run some basic Javascript templates (e.g. [Arrays](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/aba42a07da0d7f0645bc487b9a436690b4b5a1c7/scripts/generate/templates/Arrays.js)).

Since we need to make sure that the generated contracts are in sync with the templates, we run `test:generation` to make sure we're indeed in sync